llvm/lib/Analysis/VFABIDemangling.cpp

```diff
@@ -434,22 +434,33 @@ std::optional<VFInfo> VFABI::tryDemangleForVFABI(StringRef MangledName,
   // need to make sure that the VF field of the VFShape class is never
   // set to 0.
   if (IsScalable) {
-    const Function *F = M.getFunction(VectorName);
-    // The declaration of the function must be present in the module
-    // to be able to retrieve its signature.
-    if (!F)
+    if (ISA != VFISAKind::SVE)
       return std::nullopt;
-    const ElementCount EC = getECFromSignature(F->getFunctionType());
```

Comment:
What about keeping `getECFromSignature` but passing it the `ISA` and the `FunctionType` of the `CallInst`?

Reply:
But to get the FunctionType, I need the Function itself, and to get the Function I need the Module. And what about an Instruction that isn't a CallInstruction and won't give me a Function? Or do you mean passing FunctionType instead of the vector with Operand Types? Or passing the Instruction itself as argument?

Reply:
Resolved.